dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/EndpointDefinitions.java

```diff
@@ -94,7 +95,24 @@ public EndpointDefinitions(ErrorContext errorContext, Elements elements, Types t
     private Optional<HttpPath> getHttpPath(Element context, Request requestAnnotation) {
         try {
             UriTemplateParser uriTemplateParser = new UriTemplateParser(requestAnnotation.path());
-            return Optional.of(ImmutableHttpPath.of(uriTemplateParser.getNormalizedTemplate()));
+
+            Splitter splitter = Splitter.on('/');
+            Iterable<String> rawSegments = splitter.split(uriTemplateParser.getNormalizedTemplate());
+
+            List<HttpPathSegment> pathSegments = new ArrayList<>();
+            for (String segment : rawSegments) {
+                if (segment.isEmpty()) {
+                    continue; // avoid empty segments; typically the first segment is empty
+                }
+
+                if (segment.startsWith("{") && segment.endsWith("}")) {
+                    pathSegments.add(HttpPathSegments.variable(segment.substring(1, segment.length() - 1)));
```

Comment:
same as the other PR, I imagine we'll want to validate that this doesn't use any of the funky regex options e.g. `/foo/{var:.*}`
No change requested to move this PR forward though.

Reply:
Ack, added to tracking doc.